engine/cdn/websocket.go

```diff
@@ -109,9 +117,15 @@ func (s *Service) websocketOnMessage(e sdk.CDNWSEvent) {
 
 	for _, id := range clientIDs {
 		c := s.WSServer.GetClientData(id)
-		if c == nil || c.itemFilter == nil || !(c.itemFilter.ItemType == e.ItemType && c.itemFilter.APIRef == e.APIRef) {
+		if c == nil || c.itemFilter == nil || c.itemFilter.JobRunID != e.JobRunID {
 			continue
 		}
+		if e.NewItemUnitID != "" {
```

Comment:
If the item unit id is always set for each event, then create a new entry in c.itemUnitsData only when missing.